plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -546,6 +697,22 @@ int process_and_send(struct flb_cloudwatch *ctx, struct cw_flush *buf,
     int found = FLB_FALSE;
     struct flb_time tms;
 
+    /* Added for EMF support */
+    struct flb_ir_metric *metric;
+
+    int ir_metric_type;
+    char *ir_metric_unit;
+    if (strncmp(input_plugin, "cpu", INPUT_PLUGIN_CPU_LENGTH) == 0) {
+        ir_metric_type = GAUGE;
+        ir_metric_unit = PERCENT;
+    } else if (strncmp(input_plugin, "mem", INPUT_PLUGIN_MEM_LENGTH) == 0) {
+        ir_metric_type = GAUGE;
+        ir_metric_unit = BYTES;
+    } else {
+        flb_plg_info(ctx->ins, "Incompatible metric input.");
+        return -1;
```

Comment:
Why return here? If the input plugin is not for metrics.. then don't you want to continue and process the msgpack as logs like normal?

Reply:
Updated.